blog/2018/12/21/a-little-christmas-template-cooking/index.markdown

```diff
@@ -0,0 +1,104 @@
+data:
+  bio: briandfoy
+  description:
+---
+The Advent Calendar has shown you many great ways to use Mojolicious, and since you already have Mojo installed you can use it for things besides web processing. Today's recipe uses The templating rendering engine for something other than web responses.
+---
+
+First, process some string templates. Here's an example lifted from the [Mojo::Template](https://mojolicious.org/perldoc/Mojo/Template), using the [squiggly heredoc syntax released in v5.26](https://www.effectiveperlprogramming.com/2016/12/strip-leading-spaces-from-here-docs-with-v5-26/)
+
+	use Mojo::Template;
+
+	my $mt = Mojo::Template->new;
+
+	say $mt->render(<<~'EOF');
+		% use Time::Piece;
+		<div>
+		  % my $now = localtime;
+		  Time: <%= $now->hms %>
+		</div>
+		EOF
+
+The lines with leading percent sings are Perl code. One of those lines loads a module, [Time::Piece](), and the other creates the variable `$now`. The `<%= %>` insert values inline. You can figure out the other template syntax on your own; it's all in the [module documentation](https://mojolicious.org/perldoc/Mojo/Template).
+
+You can invert that so that the source of the values comes from outside of the template. The `vars()` method turns on your ability to pass a hash to the template; the hash keys turn into values. Sometimes this is preferable to having too much logic in the presentation layer:
+
+	use v5.26;
+	use Mojo::Template;
+
+	my $mt = Mojo::Template->new->vars(1);
+
+	use Time::Piece;
+	my $now = localtime;
+
+	say $mt->render(<<~'EOF', { time => $now->hms } );
+		<div>
+		  Time: <%= $time %>
+		</div>
+		EOF
+
+It's just as easy to take that template from a file (or many files). This is the sort of thing I used to do this with [Template Toolkit](http://template-toolkit.org), a very fine and capable module that's as good as it ever was. But, I'm already using Mojo for quite a few things and it already has a templating engine. I can reduce my dependency count and focus on one templating language.
+
+Typically, Mojo templates use the extension _.ep_. Loop through all of the files that you specify on the command line and cook the ones that have that extension
+
+	use v5.14;
+	use Mojo::Template;
+
+	my $mt = Mojo::Template->new->vars(1);
+
+	use Time::Piece;
+	my $now = localtime;
+
+	foreach my $file ( @ARGV ) {
+		my $new_file = $file =~ s/\.ep\z//r;
+		open my $fh, '>:utf8', $new_file or do {
+			warn "Could not open $new_file: $!\n";
+			next;
+			};
+
+		print {$fh} $mt->render_file($file, { time => $now->hms } );
+		}
+
+Now that you know everything about Mojo templates, you can process a directory full of them to start a new project.
+
+	use File::Find qw(find);
+
+	use v5.14;
+	use Mojo::Template;
+
+	my $mt = Mojo::Template->new->vars(1);
+
+	use Time::Piece;
+	my $now = localtime;
+
+	my $wanted = sub {
```

Comment:
To expand on the previous comment, here you could use Mojo::File::list_tree and Mojo::Collection::grep to get the files, then Mojo::File::sibling to target the output file. Again, up to you and depends entirely how much of the Mojo ecosystem you want to work in vs how much you'd just rather use existing tooling. For me I don't assume Mojo-ers necessarily know things like File::Find or even open layers. Then again, perl has the long history and perhaps I don't give people enough credit.

Reply:
Just going to add a note. I hope you don't mind.